sdk/scheduler/src/test/java/com/mesosphere/sdk/storage/MemPersisterTest.java

```diff
@@ -181,6 +181,59 @@ public void testDeleteRoot() throws Exception {
         testDeleteRootForPersister(CuratorPersister.newBuilder(mockServiceSpec).disableLock().build(), "/");
     }
 
+    @Test
```

Comment:
It would make sense to have this test exactly match the one in CuratorPersisterTest, just to validate that the two behave identically

Reply:
The only difference is curator persiter works even if the data is null. Mempersister does not. This is same for other methods as well. Should that be the case ?

Reply:
If it was like that already, I think that's fine for now.

Reply:
copy pasting my comment from earlier
we don't need to check if source path and destination path are equal because we check the following : 
- that source path is non emtpy
- that dest path is empty

Reply:
OK, I've had a look at the two examples again, and you're right that the current checks also fail in the case where the arguments are the same. With that said, the error messages shown are a little cryptic and it may make it easier to debug if we have clearer messages.

I would also add unit tests to explicitly check this case to ensure that this behaviour is maintained in future.